### PR TITLE
[keymap] init thystips's keymap from Jonavin's keymap

### PR DESCRIPTION
**Pull request by thystips**
## Description

Add custom keymap for gmmk pro rev 1 ainsi

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

None

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/gmmk/pro/rev1/ansi/keymaps/thystips/config.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+#    define RGB_MATRIX_FUNCTION_COLOR RGB_GOLD
+#    define RGB_MATRIX_KEYPRESSES RGB_MATRIX_FRAMEBUFFER_EFFECTS
+#endif
+
+#define EMOTICON_ENABLE
+
+#ifdef ENCODER_ENABLE
+#    ifndef DYNAMIC_KEYMAP_LAYER_COUNT
+#        define DYNAMIC_KEYMAP_LAYER_COUNT 4 // default in case this is not already defined elsewhere
+#    endif
```
Comment (waffle87) [inline keyboards/gmmk/pro/rev1/ansi/keymaps/thystips/config.h:43]: Default core behaviour
```suggestion
```

File: keyboards/gmmk/pro/rev1/ansi/keymaps/thystips/keymap.c [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    ),
+
+   [_RAISE] = LAYOUT(
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,          _______,
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______,
+        _______, _______, _______,                            _______,                            _______, _______, _______, _______, _______, _______
+    ),
```
Comment (waffle87) [inline keyboards/gmmk/pro/rev1/ansi/keymaps/thystips/keymap.c:68]:
```suggestion
    )
```

File: keyboards/gmmk/pro/rev1/ansi/keymaps/thystips/rgb_matrix_map.h [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
```
Comment (waffle87) [inline keyboards/gmmk/pro/rev1/ansi/keymaps/thystips/rgb_matrix_map.h:17]:
```suggestion
#pragma once

```

File: users/thystips/config.h [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
```
Comment (waffle87) [inline users/thystips/config.h:17]: This file should be removed, as it's just a duplicate of `keyboards/gmmk/pro/rev1/ansi/keymaps/thystips/config.h`. Or the other way around.

File: users/thystips/thystips.h [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+// DEFINE MACROS
+#define ARRAYSIZE(arr) sizeof(arr) / sizeof(arr[0])
```
Comment (waffle87) [inline users/thystips/thystips.h:22]: Why this over the [built in](https://github.com/qmk/qmk_firmware/blob/master/quantum/util.h#L47) macro?

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/gmmk/pro/rev1/ansi/keymaps/thystips/rules.mk [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+TD_LSFT_CAPSLOCK_ENABLE = yes
+IDLE_TIMEOUT_ENABLE = yes
+STARTUP_NUMLOCK_ON = yes
+ENCODER_DEFAULTACTIONS_ENABLE = yes
+ALTTAB_SCROLL_ENABLE = yes
+
+INVERT_NUMLOCK_INDICATOR = yes
+
+RGB_MATRIX_FUNCTION_COLOR = RGB_GOLD # See https://docs.qmk.fm/#/feature_rgb_matrix?id=colors
```
Comment (drashna) [inline keyboards/gmmk/pro/rev1/ansi/keymaps/thystips/rules.mk:14]: this doesn't appear to actually do anything.
```suggestion
```

File: keyboards/gmmk/pro/rev1/ansi/keymaps/thystips/rules.mk [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+RGB_MATRIX_KEYPRESSES = yes
+RGB_MATRIX_FRAMEBUFFER_EFFECTS = yes
+
+ENCODER_MAP_ENABLE = no # See https://docs.qmk.fm/#/feature_encoders?id=encoder-map
```
Comment (drashna) [inline keyboards/gmmk/pro/rev1/ansi/keymaps/thystips/rules.mk:19]:
```suggestion
ENCODER_MAP_ENABLE = no
```

File: users/thystips/thystips.c [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+            if (record->event.pressed) SEND_STRING("(~_~;)");
+            break;
+        case EMO_JOY:
+            if (record->event.pressed) SEND_STRING("(^o^)");
+            break;
+        case EMO_SAD:
+            if (record->event.pressed) SEND_STRING(":'-(");
+            break;
+        case EMO_TFLIP:
+            if (record->event.pressed) SEND_STRING("(ノಠ痊ಠ)ノ彡┻━┻");
```
Comment (drashna) [inline users/thystips/thystips.c:185]: This won't work, as it's attempting to send unicode, which isn't supported.

**Review by drashna — CHANGES_REQUESTED** [review]

File: users/thystips/thystips.c (hunk 8 above)
Comment (thystips) [reply users/thystips/thystips.c]: This doesn't work very well on linux and I don't use it.

**Comment by thystips** [comment]
@drashna Tanks for review

**Comment by zvecr** [comment]
Unfortunately since this PRs creation, QMK has [changed its stance](https://docs.qmk.fm/#/ChangeLog/20230827?id=notable-changes) on accepting user keymaps to the repository.

> qmk/qmk_firmware is no longer accepting PRs for keymaps other than for manufacturer-supported keymaps. User keymap workflow has been documented [here](https://docs.qmk.fm/#/newbs) for several years. This change is to progressively reduce the maintenance burden on the project, and to allow us to focus on the core features of QMK.

In the near future we will be providing a more user focused workflow for keymaps, but till then the suggestion is to keep changes in your own fork.

While we appreciate the time that was taken on this PR, hopefully you understand we are closing it, so that QMK can improve in the long term.

**Pull request closed without merging**